lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,309 @@
+                 'json.pretty': 1}
+            return get_response(deploy_content_standalone)
+    else:
+        return response_add_content
+
+
+def get_response(request_data):
+    headers = {'Content-Type': 'application/json'}
+    response = ''
+    try:
+        response = open_url(
+            controller_address + '/management',
+            follow_redirects=None,
+            method='POST',
+            headers=headers,
+            url_username=username,
+            url_password=password,
+            force_basic_auth=False,
+            data=json.dumps(request_data),
+        )
+    except urllib_error.HTTPError:
+        e = get_exception()
+        return e
+
+    return response
+
+
+def is_deployed(deployment):
+    deployment = '{}'.format(deployment)
+    request_data = {'operation': 'read-resource',
+                    'address': ['deployment', deployment],
+                    'json.pretty': 1}
+    return get_response(request_data)
+
+
+def deploy_disable(result, deployment, server_groups):
+    if result.get('outcome') == 'failed':
+        module.exit_json(changed=False, stdout='', stderr='')
+    elif result.get('outcome') == 'success':
+        undeploy_result = undeploy(deployment, server_groups)
+        if undeploy_result.code is 200:
+            module.exit_json(changed=True, stdout='undeployed',
+                             stderr='')
+        else:
+            module.fail_json(msg='Failed to undeploy',
+                             stdout=undeploy_result.read(),
+                             stderr=undeploy_result.read())
+
+
+def deploy_enable(result, deployment, server_groups, src):
+
+    if result.get('outcome') == 'failed':
+        deployment_result = deploy(deployment, server_groups, src)
+        if deployment_result.code is 200:
+            module.exit_json(changed=True, stdout='deployed', stderr='')
+        else:
+            module.fail_json(msg='Failed to deploy',
+                             stdout=deployment_result.read(),
+                             stderr=deployment_result.read())
+    elif result.get('outcome') == 'success':
+        undeploy_result = undeploy(deployment, server_groups)
+        if undeploy_result.code is 200:
+            deployment_result = deploy(deployment, server_groups, src)
+            if deployment_result.code is 200:
+                module.exit_json(changed=True, stdout='deployed',
+                                 stderr='')
+            else:
+                module.fail_json(msg='Failed to deploy',
+                                 stdout=deployment_result.read(),
+                                 stderr=deployment_result.read())
+        else:
+            module.fail_json(msg='Failed to undeploy',
+                             stdout=undeploy_result.read(),
+                             stderr=undeploy_result.read())
+
+
+def main():
+
+    global module, mode, changed, username, password, controller_address
```

Comment:
I don't like this. You should pass the `module.params` to the functions instead of making variables global.